### ISSUE TITLE
Make climbing easier

### ISSUE THREAD

**Issue by Alzter** [SuperTux/supertux#972]
**SuperTux version: 0.6.0 Alpha 3**
**System information:** *Windows 10*

I believe that climbing physics are really hard to use at the moment, so I have some suggestions for improvements.

1. Don't make Tux fall off the ladder or climbable object when reaching the edge of the climbable area, but instead just have the edge as a boundary. I keep slipping off climbables like this and it's a problem because in order to jump at the maximum height off a climbable surface, you have to carefully navigate to the top of it but don't go too high as you'd fall off, and then jump. It's inconvenient.

2. Remove action button letting go of the ladder. I constantly hold down action all the time because I am so used to run buttons, and I feel that having it take you off the ladder or climbable object is unneccesary when you can just jump off it with jump anyway.

3. Increase jump height and momentum upon jumping off a climbable. At the moment when Tux jumps off a climbable it's hard just to get up to platforms above it because if you keep holding up you fall off with no real upward velocity and jumping doesn't get you to the platform unless you're at the very top of the ladder. I suggest as well as removing falling off climbables (see point 1 & 2), and make Tux gain much more height when jumping off a climbable. Of course Tux should only get the maximum amount of height from holding jump, just tapping it should only send him a short distance up. And finally please increase Tux's horizontal movement speed upon jumping off a ladder so he can navigate easier.

(TL;DR: Make it identical to NSMB)

**Comment by HybridDog** [comment]
1.
I agree. If you hold Up, Tux climbs to the top and then suddenly jumps and quickly falls down until you release and press Up again.

2.
When there is an enemy above you, you may want to fall down and not jump into it. You could make action do nothing but make Tux fall without jumping when pressing the jump button while holding Down.

3.
I disagree. Jumping from a ladder is unrealistic and a ladder should lean against something. If you could jump high from it, in some levels it may be e.g. placed so that it does not lean against a tree branch but ends somewhere under it. In my opinion, jumping low from a ladder is a neat feature.

**Comment by Alzter** [comment]
> Jumping from a ladder is unrealistic and a ladder should lean against something. If you could jump high from it, in some levels it may be e.g. placed so that it does not lean against a tree branch but ends somewhere under it. In my opinion, jumping low from a ladder is a neat feature.

Just because something is realistic doesn't make it fun. Sure, you aren't able to jump high off a ladder in real life, but SuperTux isn't real life. Can you eat an egg half your size and grow twice as tall in real life? No! So I don't understand why all of the sudden everything has to be realistic. In terms of gameplay, the ladder jump is really low and makes it hard to jump from ladders to platforms, and I think it needs changing.

**Comment by HybridDog** [comment]
I thought that jumping high from a ladder could be a bit boring because it's the same you can do in Super Mario games, and gameplay features which differ from popular existing games (such as willowisps) could make the game more interesting for people who have played lots of platform games before.

**Comment by Alzter** [comment]
> I thought that jumping high from a ladder could be a bit boring because it's the same you can do in Super Mario games, and gameplay features which differ from popular existing games (such as willowisps) could make the game more interesting for people who have played lots of platform games before.

Having the game be similar to Mario is good. Mario is the standard level of quality platformers. Every time SuperTux tries to deviate from that (zeeklings, owls, tiny ladder jumps) it usually ends up worse.

**Comment by Wuzzy2** [comment]
I agree with all suggestions. I really hate latters so much. If you play with a D-pad, it's also very easy to slip off if you press left or right too long.

Especially the tiny jump height is really annoying. This is because the platform you need to jump to is often placed in such a way that you need to make a near-pixel-perfect jump to make it. In 80% of cases, I fail.

But maybe this problem can be solved w/o increasing jump height.

Remember the first level with a ladder (A Fork in the Road)? If you just climb to the top and press jump, you don't drop on the top of the ice block as you would expect, NO! You will fall all the way down again. This just doesn't make any sense. It seems the ladder is actually a hole in the ice block. If you are on top of the ice block (standing!) and walk back to the ladder, you just fall down.

I think fixing the ladder problems requires also an update of all levels with ladders to playtest them that they are not annoying to use.

All Mario arguments are irrelevant here. What matters is what makes a good game, not how well ST is cloning Mario.

**Comment by Wuzzy2** [comment]
Oops. I think I need to give more details about OP's suggestions:

Suggestion 1: Yes, the slipping off is easily the most frustrating part of ladders. This already will improve ladders by a factor of 10.
Suggestion 2: Actually, I am not that sure if this is a good idea. I am not convinced tbh. You cannot run on ladders. Ladders are supposed to be slow.
Suggestion 3: I would approve of this suggestion normally, but I also recognize the opposition to that. That's why I have suggested an alternative idea to increasing jump height. But the goal of this suggestion is right, for making it less painful to reach platforms.

**Comment by HybridDog** [comment]
> If you are on top of the ice block (standing!) and walk back to the ladder, you just fall down.

I can confirm this.
![lad](https://user-images.githubusercontent.com/3192173/71441001-cdd49b00-26ff-11ea-99d1-1747d3307d0c.jpg)
![ladcol](https://user-images.githubusercontent.com/3192173/71441002-cdd49b00-26ff-11ea-9fba-091ddb4cdd2b.jpg)

Walkable ladder pieces are a neat idea in my opinion. With these tiles you could completely disable jumping from a ladder (i.e. just fall down after pressing space). After landing on the walkable ladder tile you could jump off to other places or press up to attach to the ladder and then climb back down.

**Comment by Rusty-Box** [comment]
Point 3 has been fixed. Only the other two remain.

**Comment by tobbi** [comment]
I might tackle no. 1 today. So basically, you're saying that Tux can't exit the climbing state by reaching the borders of the climbable area alone. He should simply remain there until someone presses jump or action, right?

**Comment by Alzter** [comment]
> I might tackle no. 1 today. So basically, you're saying that Tux can't exit the climbing state by reaching the borders of the climbable area alone. He should simply remain there until someone presses jump or action, right?

Yeah, maybe just make jumping take you off the ladder though, I don't see where it'd be needed to fall off the ladder without jumping.

**Comment by Zwatotem** [comment]
> Yeah, maybe just make jumping take you off the ladder though, I don't see where it'd be needed to fall off the ladder without jumping.

There is such a case. If something is falling on you, you want to run away straight downwards. I was thinking of (if you don't want action to trigger falling) using the arrow keys, to point what direction Tux should jump, particularly `jump` + `down` means let off the ladder.

**Comment by Zwatotem** [comment]
#1670 Takes care of point 1, at least vertically (I think this is the main point of concern anyway).

**Comment by weluvgoatz** [comment]
Is this still an issue? #3 seems to have been totally resolved, #2 as well, and #1 is as Zwatotem says.

**Comment by mrkubax10** [comment]
Doesn't seem like an issue anymore.

**Comment by Alasdairbugs** [comment]
Fixed as of 0.6.3
